src/V3Timing.cpp

```diff
@@ -0,0 +1,1190 @@
+//          If fork is not join_none:
+//              Create a join struct used for syncing processes
+//
+//      Each EventControl, Wait:
+//          Create event variables for triggering those.
+//          For Wait:
+//              Transform into:
+//                  while (!wait.condp)
+//                      @(vars from wait.condp);
+//                  wait.bodysp;
+//              (process the new EventControl as specified below)
+//          For EventControl:
+//              If waiting on event, leave it as is
+//              If waiting on posedge:
+//                  Create a posedge event variable for the awaited signal
+//              If waiting on negedge:
+//                  Create a negedge event variable for the awaited signal
+//              If waiting on bothedge:
+//                  Split it into posedge and negedge
+//                  Create a posedge event variable for the awaited signal
+//                  Create a negedge event variable for the awaited signal
+//              If waiting on anyedge:
+//                  Create a anyedge event variable for the awaited signal
+//              For each variable in the condition being waited on:
+//                  Create an anyedge event variable for the awaited variable
+//
+//      Each continuous assignment:
+//          If there is an edge event variable associated with the LHS:
+//              Transform it into an Always with a normal assign (needed for next step)
+//
+//      Each assignment:
+//          If there is an edge event variable associated with the LHS:
+//              Create an EventTrigger for this event variable under an if that checks if the edge
+//              occurs
+//      Each clocked var:
+//          If there is an edge event variable associated with it:
+//              Create a new Active for this edge with an EventTrigger for this event variable
+//
+//*************************************************************************
+
+#include "config_build.h"
+#include "verilatedos.h"
+
+#include "V3Global.h"
+#include "V3Timing.h"
+#include "V3Ast.h"
+#include "V3EmitCBase.h"
+#include "V3UniqueNames.h"
+
+//######################################################################
+// Transform intra assignment delays
+
+class TimingIntraAssignControlVisitor final : public VNVisitor {
+private:
+    // STATE
+    AstScope* m_scopep = nullptr;  // Current scope
+    V3UniqueNames m_intraVarNames;  // Temp var name generator
+    bool m_underFork = false;  // Are we under a fork?
+
+    // METHODS
+    AstVarScope* getCreateIntraVar(AstNode* nodep) {
+        auto name = "__Vintraval__" + m_intraVarNames.get(nodep);
+        AstNodeModule* modp = m_scopep->modp();
+        auto* const varp
+            = new AstVar{nodep->fileline(), VVarType::BLOCKTEMP, name, nodep->dtypep()};
+        modp->addStmtp(varp);
+        AstVarScope* varscp = new AstVarScope{nodep->fileline(), m_scopep, varp};
```

Comment:
```suggestion
        AstVarScope* const varscp = new AstVarScope{nodep->fileline(), m_scopep, varp};
```

Check rest of patch likewise for const.